ltm/scratchpad.py

```diff
@@ -49,6 +49,10 @@
 This is what the user information looks like now:
 {user_info}
 
+Try to answer these questions:
+- Does the message contain information that will be useful in future?
+- Is the information unimportant general knowledge, or useful user specific knowledge?
```

Comment:
Maybe this part of the prompt is making it harder for the LLM to follow the environment's interface. In this step, it should really just answer 'yes' or 'no'. The idea sounds good though, so maybe adding a previous step in which you ask the agent this question, and right after it replies you can ask it the yes/no question.